multi_physics/QED/python_bindings/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+set(name pxr_qed)
+
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
+find_package(pybind11 CONFIG REQUIRED)
+
+pybind11_add_module(${name} "${name}.cpp")
+
+target_link_libraries(${name} PRIVATE PXRMP_QED)
+
+option(
+    PXRQEDPY_DOUBLE_PRECISION
+    "Compile pxr_qed python module in double precision" ON)
+
+if(PXRQEDPY_DOUBLE_PRECISION)
+    target_compile_definitions(${name} PRIVATE PXRQEDPY_DOUBLE_PRECISION=1)
+endif()
+
+set(PXRQEDPY_UNITS_VALUES SI NORM_OMEGA NORM_LAMBDA HEAVISIDE_LORENTZ)
+set(PXRQEDPY_UNITS SI CACHE STRING "Unit system for the pxr_qed python module (SI/NORM_OMEGA/NORM_LAMBDA/HEAVISIDE_LORENTZ)")
+set_property(CACHE PXRQEDPY_UNITS PROPERTY STRINGS ${PXRQEDPY_UNITS_VALUES})
+
+if(NOT PXRQEDPY_UNITS IN_LIST PXRQEDPY_UNITS_VALUES)
+    message(FATAL_ERROR "PXRQEDPY_UNITS (${PXRQEDPY_UNITS}) must be one of ${PXRQEDPY_UNITS_VALUES}")
+endif()
+
+if(PXRQEDPY_UNITS STREQUAL "SI")
+    target_compile_definitions(${name} PRIVATE PXRQEDPY_SI=1)
+elseif(PXRQEDPY_UNITS STREQUAL "NORM_OMEGA")
+    target_compile_definitions(${name} PRIVATE PXRQEDPY_NORM_OMEGA=1)
+elseif(PXRQEDPY_UNITS STREQUAL "NORM_LAMBDA")
+    target_compile_definitions(${name} PRIVATE PXRQEDPY_NORM_LAMBDA=1)
+elseif(PXRQEDPY_UNITS STREQUAL "HEAVISIDE_LORENTZ")
+    target_compile_definitions(${name} PRIVATE PXRQEDPY_HEAVISIDE_LORENTZ=1)
+else()
+    message(FATAL_ERROR "PXRQEDPY_UNITS is incorrectly defined")
+endif()
+
+set_target_properties(${name} PROPERTIES
+                              INTERPROCEDURAL_OPTIMIZATION ON)
+
+# OpenMP support
+if(PXRMP_QED_OMP)
+   find_package(OpenMP REQUIRED)
+   target_link_libraries(${name} PRIVATE OpenMP::OpenMP_CXX)
```

Comment:
If you link `PUBLIC` in `PXRMP`'s target against OpenMP and then link `${name}` against the PXRMP target then all those dependencies are automatically forwarded and don't need to be added again.

Reply:
I've tried, but `cmake` gives me the following error :cry:  :
```
   INTERFACE library can only be used with the INTERFACE keyword of
   target_link_libraries

```

Reply:
have you tried `target_link_libraries(${name} INTERFACE ...)` ? :)
https://stackoverflow.com/questions/26037954/cmake-target-link-libraries-interface-dependencies

Reply:
Do you mean in the `CMakeLists.txt` in `python_bindings`? I've tried to change `target_link_libraries(${name} PRIVATE PXRMP_QED)` into`target_link_libraries(${name} INTERFACE PXRMP_QED)` , but compilation of the python bindings fails, with an error message stating that `picsar_qed/math/vec_functions.hpp` does not exist...

Reply:
In that case it might be possible that the `picsar_qed/` path is not in the `PXRMP_QED`'s target interface include libraries.

Property: https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html (`target_include_directories`)

Reply:
looking at your source I cannot spot it immediately either, we can refine this in a follow-up so that the forwarding works without adding the same public library dependencies again :)